test/e2e/replica_set_custom_persistent_volume/replica_set_custom_persistent_volume.go

```diff
@@ -0,0 +1,141 @@
+package replica_set_custom_persistent_volume
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+
+	v1 "github.com/mongodb/mongodb-kubernetes-operator/api/v1"
+	"github.com/mongodb/mongodb-kubernetes-operator/test/e2e/mongodbtests"
+	. "github.com/mongodb/mongodb-kubernetes-operator/test/e2e/util/mongotester"
+	"github.com/stretchr/testify/assert"
+
+	e2eutil "github.com/mongodb/mongodb-kubernetes-operator/test/e2e"
+	setup "github.com/mongodb/mongodb-kubernetes-operator/test/e2e/setup"
+	corev1 "k8s.io/api/core/v1"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestMain(m *testing.M) {
+	code, err := e2eutil.RunTest(m)
+	if err != nil {
+		fmt.Println(err)
+	}
+	os.Exit(code)
+}
+
+func getPersistentVolumeLocal(name string, localPath string, label string) corev1.PersistentVolume {
+	return corev1.PersistentVolume{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: e2eutil.OperatorNamespace,
+			Labels:    map[string]string{"type": label},
+		},
+		Spec: corev1.PersistentVolumeSpec{
+			PersistentVolumeSource: corev1.PersistentVolumeSource{
+				Local: &corev1.LocalVolumeSource{
+					Path: localPath,
+				},
+			},
+			AccessModes:      []corev1.PersistentVolumeAccessMode{"ReadWriteOnce"},
+			StorageClassName: "default",
+			NodeAffinity: &corev1.VolumeNodeAffinity{
+				Required: &corev1.NodeSelector{
+					NodeSelectorTerms: []corev1.NodeSelectorTerm{
+						{
+							MatchExpressions: []corev1.NodeSelectorRequirement{
+								{
+									Key:      "kubernetes.io/hostname",
+									Operator: "In",
+									Values:   []string{"kind-control-plane"},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func getVolumes(ctx *e2eutil.Context, volumeType string, members int) []corev1.PersistentVolume {
+	volumes := make([]corev1.PersistentVolume, members)
+	for i := 0; i < members; i++ {
+		volumes[i] = getPersistentVolumeLocal(
+			fmt.Sprintf("%s-volume-%d", volumeType, i),
+			fmt.Sprintf("/opt/data/mongo-%s-%d", volumeType, i),
+			volumeType,
+		)
+	}
+	return volumes
+}
+
+func getPvc(pvcType string, mdb v1.MongoDBCommunity) corev1.PersistentVolumeClaim {
+	name := ""
+	if pvcType == "logs" {
+		name = mdb.LogsVolumeName()
+	} else {
+		name = mdb.DataVolumeName()
+	}
+	defaultString := "default"
+	return corev1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: corev1.PersistentVolumeClaimSpec{
+			AccessModes: []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce},
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{"type": pvcType},
```

Comment:
use `MatchLabels` as a way to ensure binding to the volume that has been created separately